src/components/Dialog/__stories__/DialogShowcase.tsx

```diff
@@ -123,6 +124,90 @@ function OtherDialog() {
     );
 }
 
+function DynamicHeightDialog() {
+    const [open, setOpen] = React.useState(false);
+    const [isFirstDynamicPartOpen, setIsFirstDynamicPartOpen] = React.useState(false);
+    const [isSecondDynamicPartOpen, setIsSecondDynamicPartOpen] = React.useState(false);
+
+    const switchVisibility = () => {
+        setOpen((prevOpen) => !prevOpen);
+    };
+
+    React.useEffect(() => {
+        if (open) {
+            setTimeout(() => {
+                setIsFirstDynamicPartOpen(true);
+            }, 2000);
+
+            setTimeout(() => {
+                setIsSecondDynamicPartOpen(true);
+            }, 4000);
+        } else {
+            setIsFirstDynamicPartOpen(false);
+            setIsSecondDynamicPartOpen(false);
+        }
+    }, [open]);
```

Comment:
Here and in the Modal story:
```suggestion
    React.useEffect(() => {
        const timers: number[] = [];
        if (props.open) {
            timers[0] = window.setTimeout(() => {
                setIsFirstDynamicPartOpen(true);
            }, 2000);

            timers[1] = window.setTimeout(() => {
                setIsSecondDynamicPartOpen(true);
            }, 4000);
        } else {
            setIsFirstDynamicPartOpen(false);
            setIsSecondDynamicPartOpen(false);
        }
        return () => {
            timers.forEach((t) => {
                window.clearTimeout(t);
            });
        };
    }, [props.open]);
```

Reply:
done